test/gtest/conv_igemm_dynamic_xdlops.cpp

```diff
@@ -0,0 +1,221 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2023 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <tuple>
+#include <miopen/miopen.h>
+#include <gtest/gtest.h>
+#include "../conv2d.hpp"
+#include "get_handle.hpp"
+
+using TestCase = std::tuple<std::vector<std::string>, std::string>;
```

Comment:
```suggestion
using TestCase = std::string;
```

The reason is that the library reads each environment variable only once (and caches theirs values in static objects). This is important for speed as environment accesses are slow.

Therefore, once the first TestCase has executed, there is no guarantee that further changes to environment variables will be seen by the library. Thus, it does not make sense to include a different set of environment variables in each TestCase. The environment should be set only once per each gtest executable.

@junliume @JehandadKhan We need to inspect existing gtests and, possibly, make similar changes there. Please let me know if you wish me to open a ticket and/or make this work myself (however the latter may be difficult for me if updated tests start to fail on MI100/200/300)

Reply:
@atamazov , I have tried the death test, with environment variable set to static const in the APIs in src/include/miopen/env.hpp

This PR is to address this issue:
https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2514

Reply:
@xinlipn I do not like "death test" approach as well. But there are some other means to run each test as in a separate process. And #2514 is not free. Please see https://github.com/ROCmSoftwarePlatform/MIOpen/issues/2521#issuecomment-1819467431 for more info.

Reply:
@xinlipn 
> I have tried the death test,

Thanks for evaluating it.